auth/kerberos/src/krb.cpp

```diff
@@ -716,19 +718,39 @@ std::pair<int, std::string> get_gmsa_krb_ticket( std::string domain_name,
     return std::make_pair( error_code, krb_cc_name );
 }
 
+/**
+ * Parses the string that is a result of the klist command for the ticket experation date and time
+ * @param klist_ticket_info  - String output of the klist command to parse
+ * @return - returns the date and time of the ticket experation otherwise an empty string
+ */
+std::string get_ticket_expiration( std::string klist_ticket_info )
+{
+    std::regex pattern(".+\n.{21}(.{19}).+");
+    std::smatch expires_match;
+
+    if (!std::regex_search(klist_ticket_info, expires_match, pattern))
+    {
+        std::cout << "Unable to parse klist for ticket experation: " << klist_ticket_info << std::endl;
+        return "";
+    }
+
+    return expires_match[1];
```

Comment:
Maybe use a new string to copy expires_match[1] to avoid any memory issues with returning strings.

Reply:
Addressed in latest updated PR